cunumeric/array.py

```diff
@@ -4014,7 +4014,7 @@ def find_common_type(*args: Any) -> np.dtype[Any]:
                 scalar_types.append(array.dtype)
             else:
                 array_types.append(array.dtype)
-        return np.find_common_type(array_types, scalar_types)  # type: ignore
+        return np.find_common_type(array_types, scalar_types)
```

Comment:
I believe this change is causing this mypy error in ci: https://github.com/nv-legate/cunumeric/actions/runs/5447445212/jobs/9911480442?pr=925#step:10:644

Reply:
This is from #980 . I'll undo these changes once @ipdemes merges #980 .